bundles/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/MeterHandler.java

```diff
@@ -0,0 +1,253 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pixometer.handler;
+
+import static org.openhab.binding.pixometer.internal.PixometerBindingConstants.*;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.quantity.Energy;
+import javax.measure.quantity.Volume;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.pixometer.internal.PixometerMeterConfiguration;
+import org.openhab.binding.pixometer.internal.config.ReadingInstance;
+import org.openhab.binding.pixometer.internal.serializer.CustomReadingInstanceDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link MeterHandler} is responsible for handling data and measurements of a meter thing
+ *
+ * @author Jerome Luckenbach - Initial contribution
+ */
+public class MeterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MeterHandler.class);
+
+    private static final String API_VERSION = "v1";
+    private static final String API_METER_ENDPOINT = "meters";
+    private static final String API_READINGS_ENDPOINT = "readings";
+
+    private final GsonBuilder gsonBuilder = new GsonBuilder().registerTypeAdapter(ReadingInstance.class,
+            new CustomReadingInstanceDeserializer());
+    private final Gson gson = gsonBuilder.create();
+
+    private String resourceID;
+    private String meterID;
+
+    private ScheduledFuture<?> pollingJob;
+
+    public MeterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No handling needed, since we have only values to read.
```

Comment:
My code is mainly inspired from tankerkoenig and airquality bindings.
I didn't see such an implementation on tankerkoenig but found the following in airquality now:
https://github.com/openhab/openhab2-addons/blob/master/bundles/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/handler/AirQualityHandler.java#L150

Would this be the right approach here too?

Reply:
If you have a similar method like `updateChannel` available here, yes, the same approach would make sense. Alternatively, you could use the [expiring cache](https://www.openhab.org/docs/developer/utils/tools.html#expiring-and-asynchronously-reloading-cache).

Reply:
I have the updateMeter method available which updates all channels.
In my view it doesn't make any sense to separate between these channels, because alle informations are related to the last reading value channel.

But i would have to add some duplicated code in the handle command method for now, until i have refactored the api implementation.
I hope that this is ok for now.

Reply:
Note that the REFRESH command can be called for every linked channel at startup - so you must ensure that you are not calling a "update all channels" 10 times in a row.
That's where the ExpiringCache comes in handy: It internally only calls the update method, if the values are expired and otherwise serves them from the cache.

Reply:
Okay i see that expiring cache is useful here.
I have to find it in a live implementation somewhere and check it there.
The docs of that are confusing me a bit currently.

Reply:
Did you already find some suitable example? I would have to look myself, I don't know a good one by heart...

Reply:
I have searched around and think that the tplink binding implemented it nice and small,
which would fit my thing handler too.

There are several implementations with extra classes and abstract structure, but i think that would be some kind of overkill for my three available channels...
Hopefully i can do this change till the end of this week.